python/pyarrow/tests/test_substrait.py

```diff
@@ -223,7 +224,11 @@ def table_provider(names):
     assert res_tb == test_table_1
 
 
+@pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
 def test_named_table_invalid_table_name():
+    # Warning generated from pyx which is unable to be propagated to caller
+    # ref: https://docs.pytest.org/en/6.2.x/usage.html#warning-about
+    #      -unraisable-exceptions-and-unhandled-thread-exceptions
```

Comment:
Is this something we should actually try to fix?

Reply:
The ignored exception is:

```
Traceback (most recent call last):
  File "/home/joris/scipy/repos/arrow/python/pyarrow/tests/test_substrait.py", line 238, in table_provider
    raise Exception("Unrecognized table name")
Exception: Unrecognized table name
```

which is IMO something we should fix instead (the error should be raised to the user)

Reply:
That seems prudent. I'll make an issue for it separately, and have removed this filter in https://github.com/apache/arrow/pull/14729/commits/20b1a30622ecf722783c1f0485d5f37c110a8deb

Reply:
https://github.com/apache/arrow/issues/14766